File: app/src/main/java/com/polok/eubmanagement/firebase/FirebaseDataRef.java
package com.polok.eubmanagement.firebase;

import com.google.firebase.database.DatabaseReference;
import com.polok.eubmanagement.base.BaseApp;
import com.polok.eubmanagement.util.SharedPref;

public class FirebaseDataRef {
    public static DatabaseReference provideBatchRef() {
        return BaseApp.getFirebaseDataRef().child("BATCH_REF");
    }

    public static DatabaseReference provideStudentRef() {
        return BaseApp.getFirebaseDataRef().child("STUDENT").child(SharedPref.getUserBatch());
    }

    public static DatabaseReference provideScheduleRef() {
        return BaseApp.getFirebaseDataRef().child("SCHEDULE").child(SharedPref.getUserBatch());
    }

    public static DatabaseReference provideNoticeRef() {
        return BaseApp.getFirebaseDataRef().child("NOTICE").child(SharedPref.getUserBatch());
    }

    public static DatabaseReference provideAssignmentRef() {
        return BaseApp.getFirebaseDataRef().child("ASSIGNMENT").child(SharedPref.getUserBatch());
    }

    public static DatabaseReference provideModuleRef() {
        return BaseApp.getFirebaseDataRef().child("MODULE");
    }

    public static DatabaseReference provideFacultyRef() {
        return BaseApp.getFirebaseDataRef().child("FACULTY");
    }
}

File: app/src/main/java/com/polok/eubmanagement/firebase/FirebaseDataRef.kt
package com.polok.eubmanagement.firebase

import com.google.firebase.database.DatabaseReference
import com.polok.eubmanagement.base.BaseApp
import com.polok.eubmanagement.util.SharedPref.getUserBatch

object FirebaseDataRef {

    fun provideBatchRef(): DatabaseReference? {
        return BaseApp.firebaseDataRef?.child("BATCH_REF")
    }

    fun provideStudentRef(): DatabaseReference? {
        return BaseApp.firebaseDataRef?.child("STUDENT")?.child(getUserBatch())
    }

    fun provideScheduleRef(): DatabaseReference? {
        return BaseApp.firebaseDataRef?.child("SCHEDULE")?.child(getUserBatch())
    }

    fun provideNoticeRef(): DatabaseReference? {
        return BaseApp.firebaseDataRef?.child("NOTICE")?.child(getUserBatch())
    }

    fun provideAssignmentRef(): DatabaseReference? {
        return BaseApp.firebaseDataRef?.child("ASSIGNMENT")?.child(getUserBatch())
    }

    fun provideModuleRef(): DatabaseReference? {
        return BaseApp.firebaseDataRef?.child("MODULE")
    }

    fun provideFacultyRef(): DatabaseReference? {
        return BaseApp.firebaseDataRef?.child("FACULTY")
    }
}